### DISCUSSION TITLE
HTTP/3 support

### DISCUSSION THREAD

**Discussion by cherinyy** [arvidn/libtorrent#7169] [Ideas]
Hello,

The idea is mainly about using the [HTTP/3](https://datatracker.ietf.org/doc/html/rfc9114) protocol on Tracker.

currently, the prerequisite for an HTTP/3 connection is to connect to the server via HTTP/2 and then upgrade to the HTTP/3 protocol via [ALTSVC](https://datatracker.ietf.org/doc/html/rfc7838). This obviously does not apply to libtorrent.

![image](https://user-images.githubusercontent.com/9050713/197375402-bbbda3dc-c98b-4582-9317-3a9dad90cdb1.png)

```
curl -I https://www.cloudflare.com/
```

another common practice is to select protocols via [DNS SVCB and HTTPS RRs](https://datatracker.ietf.org/doc/html/draft-ietf-dnsop-svcb-https).

![image](https://user-images.githubusercontent.com/9050713/197375678-6e4e09db-2a40-41ca-a445-036be00b5503.png)

```
dig @1dot1dot1dot1.cloudflare-dns.com +dnssec +short +tcp dash.cloudflare.com HTTPS
```

or libtorrent can design its own way to support HTTP/3.

Best regards,
wordlesswind

**Comment by arvidn** [comment]
is there any demand for this?
It might be simpler to add authentication and confidentiality to the UDP tracker protocol (and probably slightly more efficient)

**Reply by cherinyy** [reply]
> is there any demand for this?

Perhaps because [privacy leakers](https://www.reuters.com/article/china-data-hack-idCAKBN2OF0I1) were using BitTorrent to share sample data, one of the Chinese government's responses was to poison the DNS of the DHT bootstrap node's domain. 

Since DHT uses UDP, the bootstrap node should work properly when I specify the correct IP address in the HOSTS file. But it doesn't, my landline still can't connect to the DHT node, even though I see node data being returned via Wireshark. Works properly with mobile networks.

I will most likely not be able to connect to the UDP Tracker, even if I can connect to the same Tracker very quickly using HTTPS or HTTP.

However, when transferring files, neither TCP nor UDP is affected, my network is really weird right now.

I want libtorrent to support HTTP/3 because it's a general-purpose protocol that isn't specifically identified as BitTorrent. it's UDP-based, which means it's faster. And it doesn't have TLS RST, which makes it impossible for a man-in-the-middle to block connections by preempting RST packets.

It looks great.
